core/src/main/scala/org/apache/spark/sql/delta/DeltaHistoryManager.scala

```diff
@@ -163,17 +163,26 @@ class DeltaHistoryManager(
       version: Long,
       mustBeRecreatable: Boolean = true,
       allowOutOfRange: Boolean = false): Unit = {
-    val earliest = if (mustBeRecreatable) {
-      getEarliestReproducibleCommit
-    } else {
-      getEarliestDeltaFile(deltaLog)
-    }
+    val earliest = getEarliestVersion(mustBeRecreatable = mustBeRecreatable)
     val latest = deltaLog.update().version
     if (version < earliest || ((version > latest) && !allowOutOfRange)) {
       throw VersionNotFoundException(version, earliest, latest)
     }
   }
 
+  /**
+   * Get the earliest delta version.
+   * @param mustBeRecreatable whether the snapshot of this version needs to be recreated.
+   */
+  def getEarliestVersion(
+      mustBeRecreatable: Boolean = true): Long = {
+    if (mustBeRecreatable) {
+      getEarliestReproducibleCommit
```

Comment:
nit: Add `()` at the end (as seems so Java-ish with `get` at the beginning)

Reply:
Adding `()` breaks the compilation so I removed them:  
![image](https://user-images.githubusercontent.com/33152084/183277943-26778627-75bc-4c9e-9f0b-58f1bec9fcc7.png)